src/session.ts

```diff
@@ -588,24 +588,24 @@ async function sendReinvite(this: WebPhoneSession, options: any = {}): Promise<a
     this.pendingReinvite = true;
     options.modifiers = options.modifiers || [];
 
-    return new Promise(async (resolve, reject) => {
-        try {
-            const description = await this.sessionDescriptionHandler.getDescription(
-                options.sessionDescriptionHandlerOptions,
-                options.modifiers
-            );
+    try {
+        const description = await this.sessionDescriptionHandler.getDescription(
+            options.sessionDescriptionHandlerOptions,
+            options.modifiers
+        );
+        const result = await new Promise((resolve, reject) => {
             this.sendRequest(C.INVITE, {
                 body: description,
                 receiveResponse: (response: IncomingResponse) => {
                     if (response.statusCode === 200) resolve(response);
-                    return this.receiveReinviteResponse(response);
                 }
             });
-        } catch (e) {
-            this.pendingReinvite = false;
-            reject(e);
-        }
-    });
+        });
+        await this.receiveReinviteResponse(result);
+        return result;
+    } catch (e) {
+        this.pendingReinvite = false;
```

Comment:
add `throw e` here to re-throw the error

Reply:
Done